### PR TITLE
Add image upload to posting

### PR DESCRIPTION
**Pull request by steam**
For some reason Google decided to disable `WebView` file uploading in Android 4.1 and later. This PR makes use of a 3rd party library, https://github.com/delight-im/Android-AdvancedWebView to restore the functionality. It should handle `<input type="file"></input>` in all versions of Android now.

This will require a lot of testing on different versions of Android to verify that it works as expected.

slack for ios upload
slack for ios upload-1

### CODE REVIEW

File: app/src/main/java/ello/co/ello/MainActivity.java [hunk 1]
```diff
@@ -84,17 +84,45 @@ private void setupNoInternetView() {
     }
 
     private void setupWebView() {
-        mWebView = (WebView) findViewById(R.id.activity_main_webview);
+        mWebView = (AdvancedWebView) findViewById(R.id.activity_main_webview);
         mWebView.setAlpha(0.0f);
+        mWebView.setListener(this, this);
+        mWebView.loadUrl("https://ello-webapp-epic.herokuapp.com");
```
Comment (jayzes) [inline app/src/main/java/ello/co/ello/MainActivity.java]: Not that it has to happen in this PR, but we should eventually set this to some sort of indirection URL (or use preview.ello.co) so was can change where it points to remotely.

**Comment by steam** [comment]
Not a bad idea @jayzes, I went ahead and changed it to https://preview.ello.co

File: app/src/main/java/ello/co/ello/MainActivity.java [hunk 2]
```diff
@@ -84,17 +84,45 @@ private void setupNoInternetView() {
...
+        mWebView.setListener(this, this);
+        mWebView.loadUrl("https://preview.ello.co");
+        mWebView.setWebViewClient(new ElloWebViewClient(this));
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent intent) {
+        super.onActivityResult(requestCode, resultCode, intent);
+        if (mWebView != null) {
+            mWebView.onActivityResult(requestCode, resultCode, intent);
```
Comment (steam) [inline app/src/main/java/ello/co/ello/MainActivity.java]: This is the little bit of magic that hands the result of selecting images back into the `WebView`. `AdvancedWebView` is super handy that way.

**Comment by steam** [comment]
ping @colinta, you mind giving this a once over sir?

File: app/src/main/java/ello/co/ello/MainActivity.java [hunk 3]
```diff
@@ -27,47 +28,46 @@ protected void onCreate(Bundle savedInstanceState) {
     }
 
     @Override public void onRefresh() {
-        mWebView.reload();
+        if(mWebView != null) {
```
Comment (colinta) [inline app/src/main/java/ello/co/ello/MainActivity.java]: Ug, these checks are for the birds!

Comment (steam) [inline app/src/main/java/ello/co/ello/MainActivity.java]: Yeah, not super fun. The issue, beyond normal java null pointer nonsense, is that `mWebView` might not be created in the case of no internet being present so any function calls on it need to be guarded.

**Comment by colinta** [comment]
I don't really understand the ins and outs, but it looks simple enough!

**Pull request merged**